File: exercism/julia/resistor-color-trio/resistor-color-trio.jl
const color_values = Dict("black" => "0", "brown" => "1", "red" => "2", "orange" => "3", "yellow" => "4",
    "green" => "5", "blue" => "6", "violet" => "7", "grey" => "8", "white" => "9")
function label(colors)
    resistance_duo_value = map(color -> color_values[color], colors[1:2]) |> join
    resistance_value = string(resistance_duo_value, repeat("0", parse(Int, color_values[colors[3]])))
    resistance_int_value = parse(Int, resistance_value)
    if resistance_int_value >= 1000 && resistance_int_value % 1000 == 0
        "$(resistance_int_value ÷ 1000) kiloohms"
    else
        "$(resistance_int_value) ohms"
    end
end

File: exercism/java/resistor-color-trio/src/main/java/ResistorColorTrio.java
import java.util.Map;

class ResistorColorTrio {
  private static final Map<String, Integer> matches = Map.of(
          "black", 0,
          "brown", 1,
          "red", 2,
          "orange", 3,
          "yellow", 4,
          "green", 5,
          "blue", 6,
          "violet", 7,
          "grey", 8,
          "white", 9
  );

  String label(String[] colors) {
    String result = matches.get(colors[0]).toString() + matches.get(colors[1]).toString()
            + "0".repeat(matches.get(colors[2]));
    long value = Long.parseLong(result);
    if (value != 0 && result.endsWith("0".repeat(9))) {
      return "%d gigaohms".formatted(value / 1_000_000_000);
    } else if (value != 0 && result.endsWith("0".repeat(6))) {
      return "%d megaohms".formatted(value / 1_000_000);
    } else if (value != 0 && result.endsWith("0".repeat(3))) {
      return "%d kiloohms".formatted(value / 1_000);
    } else {
      return "%d ohms".formatted(value);
    }
  }
}
